dolly-backend-app/src/main/java/no/nav/dolly/bestilling/aareg/util/AaregMergeUtil.java

```diff
@@ -91,14 +85,30 @@ private static boolean isEqualPersonnr(Arbeidsforhold nytt, ArbeidsforholdRespon
                         .max(Comparator.comparing(id -> id)).orElse(0)
         );
 
-        nyeArbeidsforhold.forEach(arbeidforhold -> {
-            arbeidforhold.setArbeidsforholdID(Integer.toString(arbeidsforholdId.addAndGet(1)));
-            arbeidforhold.getPermisjon().forEach(permisjon ->
+        AtomicInteger arbeidsforholdId = null;
+        try {
+            arbeidsforholdId = new AtomicInteger(
+                    eksisterendeArbeidsforhold.stream()
+                            .map(ArbeidsforholdResponse::getArbeidsforholdId)
+                            .map(id -> id.replace("-", ""))
+                            .mapToInt(Integer::valueOf)
+                            .max().orElse(0)
+            );
+        } catch (NumberFormatException e) {
+            log.warn("Klarte ikke å omgjøre arbeidsforholdId til integer, bruker string verdi fra eksisterende arbeidsforhold.", e);
+        }
+
+        for (int i = 0; i < nyeArbeidsforhold.size(); i++) {
+            Arbeidsforhold nyttArbeidsforhold = nyeArbeidsforhold.get(i);
+            nyttArbeidsforhold.setArbeidsforholdID(nonNull(arbeidsforholdId)
+                    ? Integer.toString(arbeidsforholdId.addAndGet(1))
+                    : eksisterendeArbeidsforhold.get(i).getArbeidsforholdId());
```

Comment:
Litt usikker på denne, jeg antok at gamle arbeidsforhold lengde = nye, men det er kanskje ikke tilfellet? Isåfall må det vel gjøres noe lurt for de strengene som blir sendt inn 😅

Reply:
Ja høres bra ut 👍